compiler/rustc_hir_typeck/src/coercion.rs

```diff
@@ -671,7 +671,7 @@ impl<'f, 'tcx> Coerce<'f, 'tcx> {
                         return Err(TypeError::Mismatch);
                     }
                 }
-                Err(traits::Unimplemented) => {
+                Err(err) if let traits::Unimplemented = err.kind() => {
```

Comment:
```suggestion
                Err(err) if matches!(err.kind(), traits::Unimplemented) => {
```
Nit: these kinds of Yoda conditions always look bad, IMO, especially when no bindings are introduced.
(There are several similar cases in this PR.)